grails-app/services/org/pih/warehouse/invoice/InvoiceService.groovy

```diff
@@ -211,18 +211,28 @@ class InvoiceService {
     }
 
     def updateItems(Invoice invoice, List items) {
+        List<InvoiceItem> currentInvoiceItems = InvoiceItem.findAllByInvoice(invoice)
+
         items.each { item ->
-            InvoiceItem invoiceItem = InvoiceItem.get(item.id)
+            InvoiceItem invoiceItem = currentInvoiceItems.find{ it.id == item?.id }
+            // update existing invoice item
             if (invoiceItem) {
-                invoiceItem.quantity = item.quantity
+                if (item.quantity > 0) {
```

Comment:
I'm not sure where to put the validation logic, but it would be good to encapsulate all validation logic in one place and check to see if the whole object is ok. However, since we're only updating quantity here I think we're ok. So nevermind.